internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -100,6 +114,154 @@ func validateSidecarMountPoints(in []manifest.SidecarMountPoint) error {
 	return nil
 }
 
+func validateSidecarDependsOn(in manifest.SidecarConfig, sidecarName string, s convertSidecarOpts) error {
+	if in.DependsOn == nil {
+		return nil
+	}
+
+	for name, status := range in.DependsOn {
+		// status must be one of < start | complete | success >
+		if !isValidStatus(status) {
+			return errInvalidDependsOnStatus
+		}
+		// essential containers must have 'start' as a status
+		if name == s.workloadName || s.sidecarConfig[name].Essential == nil || aws.BoolValue(s.sidecarConfig[name].Essential) {
+			if status != "start" {
+				return errEssentialContainerStatus
+			}
+		}
+	}
+
+	return nil
+}
+
+func validateNoCircularDependencies(s convertSidecarOpts) error {
+	used := make(map[string]bool)
+	path := make(map[string]bool)
+	dependencies, err := buildDependencyGraph(s)
+
+	// don't let nonexistent containers pass
```

Comment:
nit: We try to use complete and grammatically correct sentences in comments wherever possible!

```suggestion
	// Don't let nonexistent containers pass.
```

Reply:
Fixed comments.